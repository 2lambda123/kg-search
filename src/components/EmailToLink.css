/*
*   Copyright (c) 2018, EPFL/Human Brain Project PCO
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*/

a.kgs-email-link {
    display: inline-block;
    margin: 0;
    padding: 0 10px;
    border: 0;
    border-radius: 3px;
    background-color: transparent;
    font-weight: 500;
    color: var(--color-19);
    font-size: 22px;
    text-align: center;
    -webkit-transition: color 0.2s ease-in, -webkit-box-shadow 0.2s ease-in;
    transition: box-shadow 0.2s ease-in, color 0.2s ease-in, -webkit-box-shadow 0.2s ease-in;
    line-height: 1.5em;
}

a.kgs-email-link:hover {
    color: var(--color-20);
    -webkit-box-shadow: 3px 3px 6px var(--box-shadow-2);
    box-shadow: 3px 3px 6px var(--box-shadow-2);
}

