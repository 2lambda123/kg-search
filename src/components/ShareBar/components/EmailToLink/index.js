/*
*   Copyright (c) 2018, EPFL/Human Brain Project PCO
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*/

import React from "react";
import { withStoreStateSubscription} from "../../../withStoreStateSubscription";
import "./styles.css";

const EmailToLinkComponent = ({title = "Send search link by email", text, icon = "fa fa-envelope-o", link}) => {
  return (
    <a className="kgs-email-link" href={link} title={title}><i className={icon}></i><span>{text}</span></a>
  );
};

export const EmailToLink = withStoreStateSubscription(
  EmailToLinkComponent,
  data => ({
    link: data.application.shareEmailToLink
  })
);
