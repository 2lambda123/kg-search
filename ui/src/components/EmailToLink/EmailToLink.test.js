/*
*   Copyright (c) 2018, EPFL/Human Brain Project PCO
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*/

import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import EmailToLink from "./EmailToLink";

Enzyme.configure({ adapter: new Adapter() });

test('EmailToLink component renders initially', () => {
    const component = renderer.create(
        <EmailToLink className="test" title="title" text="text" link="link" icon="times" />
    );
  
    expect(component.toJSON()).toMatchSnapshot();
});

test('EmailToLink test className"', () => {
    const component = shallow(
        <EmailToLink className="test" title="title" text="text" link="link" icon="times" />
    );
    expect(component.hasClass("test"));
});
  
test('EmailToLink test text', () => {
    const component = shallow(
        <EmailToLink className="test" title="title" text="text" link="link" icon="times" />
    );
    expect(component.text()).toContain('text');
});

test('EmailToLink test title', () => {
    const component = shallow(
        <EmailToLink className="test" title="title" text="text" link="link" icon="times" />
    );
    expect(component.prop("title")).toEqual('title');
});

test('EmailToLink test icon', () => {
    const component = shallow(
        <EmailToLink className="test" title="title" text="text" link="link" icon="times" />
    );
    expect(component.find("svg"));
});