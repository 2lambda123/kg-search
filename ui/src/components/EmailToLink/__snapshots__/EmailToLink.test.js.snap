// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailToLink component renders initially 1`] = `
<a
  className="kgs-email-link test"
  href="link"
  title="title"
>
  <i
    className="icon"
  />
  <span>
    text
  </span>
</a>
`;
